### PR TITLE
feat: add row exists api

### PR DESCRIPTION
**Pull request by rahulKQL**
Adding a new Row exists() API in `BigtableDataClient`

### CODE REVIEW

**Review by pmakani — APPROVED** [review]

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 1]
```diff
@@ -159,6 +162,121 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
     this.stub = stub;
   }
 
+  /** Convenience method for synchronously confirm if a row exists or not.
+   *
```
Comment (dmitry-fa) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:169]: To meet javadoc style I would rephrase:
```
/**
  * Confirms synchronously if a row exists or not.
```

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -159,6 +162,121 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
...
+   * <p>Sample code:
+   *
+   * <pre>{code
+   * try (BigtableDataClient bigtableDataClient = BigtableDataClient.create("[PROJECT]", "[INSTANCE]")) {
```
Comment (dmitry-fa) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:173]: `<pre>{code`
->
`<pre>{@code`

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 3]
```diff
@@ -159,6 +162,121 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
...
+   *   boolean isRowPresent = bigtableDataClient.exists(tableId, key);
+   *
+   *   // Do something with result, for example, display a message
+   *   if(isRowPresent) {
+   *     System.out.println(key + " is present");
+   *   }
+   * } catch(ApiException e) {
+   *   e.printStackTrace();
+   * }
+   *
```
Comment (dmitry-fa) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:187]: you missed closing:
`*}</pre>`

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 4]
```diff
@@ -159,6 +162,121 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
...
+   * @throws com.google.api.gax.rpc.ApiException when a serverside error occurs
+   */
+  public boolean exists(String tableId, String rowKey) {
```
Comment (dmitry-fa) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:190]: Besides `@throws `javadoc should describe:  `@param tableId`, `@param rowKey` and `@returns`

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 5]
```diff
@@ -159,6 +162,121 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
...
+    return ApiExceptions.callAndTranslateApiException(existsAsync(tableId, rowKey));
+  }
+
```
Comment (dmitry-fa) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:193]: The same for other javadoc comments.

**Review by dmitry-fa — CHANGES_REQUESTED** [review]

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 1 above)
Comment (rahulKQL) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:169]: Thanks for the review!! Have updated the same.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 4 above)
Comment (rahulKQL) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:190]: I would have added it but the rest of the method does not include any `@param`, So following the same pattern here.

Comment (dmitry-fa) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:190]: okay

**Review by dmitry-fa — APPROVED** [review]
LGTM
